collect_app/src/main/java/org/odk/collect/android/preferences/AutoSendPreferenceMigrator.java

```diff
@@ -0,0 +1,83 @@
+package org.odk.collect.android.preferences;
+
+import android.annotation.SuppressLint;
+import android.content.SharedPreferences;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Map;
+
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND_NETWORK;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND_WIFI;
+
+/**
+ * Migrates existing autosend_wifi and autosend_network preference values to autosend
+ */
+public class AutoSendPreferenceMigrator {
+
+    private static boolean autoSendWifi = false;
+    private static boolean autoSendNetwork = false;
+    private static String autoSend = "off";
+
+    @SuppressLint("ApplySharedPref")
+    public static void migrate(SharedPreferences sharedPreferences) {
+
+        autoSendWifi = sharedPreferences.getBoolean(KEY_AUTOSEND_WIFI, false);
+        autoSendNetwork = sharedPreferences.getBoolean(KEY_AUTOSEND_NETWORK, false);
+        autoSend = sharedPreferences.getString(KEY_AUTOSEND, "off");
```

Comment:
I think we should avoid having strings that represent preferences defaults in the code (e.g. "off") since they're already so nicely held in one map and accessible through `GeneralSharedPreferences`. As I describe below, I don't actually think you need `autoSend` at all but if you did, I think using `GeneralSharedPreferences`' `get` method would be better.